docs/design/snapshot.md

```diff
@@ -0,0 +1,192 @@
+Snapshots are read-only copies of volumes (once the snapshot is activated) that
+can be used for cloning new volumes from. Its is possible to create a snapshot
+through the [Snapshot Create API](../api/api.md#create-a-snapshot) or the
+commandline client.
+
+From the command line client, you can type the following to create a snapshot
+from an existing volume:
+
+```
+$ heketi-cli volume snapshot-create <vol_uuid> [-name=<snap_name>]
```

Comment:
Do you want "--name" for consistency with other commands?

Reply:
Yes, that was the idea. However `snap_name` is something that can only be used once in a Gluster environment (where many different volumes may exist). So we'll keep the name in the Heketi DB only, and will not pass it on the `gluster` cmdline.

If there is no value in having a name, we can drop it.